recipes/mongo-c-driver/all/conanfile.py

```diff
@@ -0,0 +1,223 @@
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        to_replace_old_new = [
+            # Fix Snappy
+            {"old": "include (FindSnappy)\nif (SNAPPY_INCLUDE_DIRS)",
+             "new": "if(ENABLE_SNAPPY MATCHES \"ON\")\n  find_package(Snappy REQUIRED)"},
+            {"old": "SNAPPY_LIBRARIES", "new": "Snappy_LIBRARIES"},
+            {"old": "SNAPPY_INCLUDE_DIRS", "new": "Snappy_LIBRARIES"},
+            # Fix Openssl
+            {"old": "OPENSSL_FOUND", "new": "OpenSSL_FOUND"},
+            {"old": "OPENSSL_VERSION", "new": "OpenSSL_VERSION"},
+            {"old": "OPENSSL_CRYPTO_LIBRARY", "new": "OpenSSL_Crypto_LIBS"},
+            {"old": "OPENSSL_LIBRARIES", "new": "OpenSSL_LIBRARIES"},
+            {"old": "OPENSSL_INCLUDE_DIR", "new": "OpenSSL_INCLUDE_DIR"},
+            # Fix LibreSSL
+            {"old": "set (SSL_LIBRARIES -ltls -lcrypto)", "new": ""},
+        ]
+        for old_new in to_replace_old_new:
+            tools.replace_in_file(os.path.join(self._source_subfolder, "src", "libmongoc", "CMakeLists.txt"),
+                                  old_new["old"], old_new["new"])
+
+    @property
+    def ssl_cmake_value(self):
+        return {
+            "darwin": "DARWIN",
+            "windows": "WINDOWS",
+            "openssl": "OPENSSL",
+            "libressl": "LIBRESSL",
+        }.get(str(self.options.with_ssl), "OFF")
+
+    @property
+    def sasl_cmake_value(self):
+        return {
+            "sspi": "SSPI",
+            "cyrus": "CYRUS",
+        }.get(str(self.options.with_sasl), "OFF")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = "TRUE"
+        self._cmake.definitions["ENABLE_TESTS"] = "OFF"
+        self._cmake.definitions["ENABLE_EXAMPLES"] = "OFF"
+        self._cmake.definitions["ENABLE_SRV"] = "ON" if self.options.srv else "OFF"
+        self._cmake.definitions["ENABLE_STATIC"] = "OFF" if self.options.shared else "ON"
+        self._cmake.definitions["ENABLE_SSL"] = self.ssl_cmake_value
+        self._cmake.definitions["ENABLE_SASL"] = self.sasl_cmake_value
+        self._cmake.definitions["ENABLE_SNAPPY"] = "ON" if self.options.with_snappy else "OFF"
+        self._cmake.definitions["ENABLE_ZLIB"] = "SYSTEM" if self.options.with_zlib else "OFF"
+        self._cmake.definitions["ENABLE_ZSTD"] = "ON" if self.options.with_zstd else "OFF"
+        self._cmake.definitions["ENABLE_ICU"] = "ON" if self.options.with_icu else "OFF"
+        self._cmake.definitions["ENABLE_SHM_COUNTERS"] = "OFF"
+        self._cmake.definitions["ENABLE_BSON"] = "ON"
+        self._cmake.definitions["ENABLE_MONGOC"] = "ON"
+        if self.settings.os == "Windows":
+            self._cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = self.options.shared
```

Comment:
probably not required. mongo-c-driver seems to properly export public symbols: https://github.com/mongodb/mongo-c-driver/blob/14b0d73c1851b74d46345c8c554b8bf7aae941e1/src/libbson/src/bson/bson-macros.h#L78

Reply:
addressed in n-bes#2